symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -218,6 +248,92 @@ public V4Message send(@Nonnull String streamId, @Nonnull String message) {
+    tempFile.deleteOnExit();
+    FileUtils.copyInputStreamToFile(attachment.inputStream(), tempFile);
+    return executeAndRetry("send", () -> messagesApi.v4StreamSidMessageCreatePost(
+        streamId,
+        authSession.getSessionToken(),
+        authSession.getKeyManagerToken(),
+        message,
+        data,
+        null,
+        tempFile,
+        null
+    ));
+  }
+
+  /**
+   * Sends a message to the stream ID passed in parameter.
+   *
+   * @param stream      the stream to send the message to
+   * @param message     the message payload in MessageML
+   * @param data        the data to be send with the message
+   * @param attachment  the input stream of the attachment of the message
+   * @return a {@link V4Message} object containing the details of the sent message
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
+   */
+  public V4Message send(@Nonnull V4Stream stream, @Nonnull String message, String data, Attachment attachment)
+      throws IOException {
+    return send(stream.getStreamId(), message, data, attachment);
+  }
+
+  /**
+   * Sends a message to the stream ID passed in parameter.
+   *
+   * @param streamId    the ID of the stream to send the message to
+   * @param message     the message to send to the stream
+   * @return a {@link V4Message} object containing the details of the sent message
+   * @throws IOException if the attachment file cannot be created.
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
+   */
+  public V4Message send(@Nonnull String streamId, @Nonnull Message message) throws IOException {
+    File tempFile;
+    Attachment attachment = message.attachment();
+    if (attachment != null) {
+      tempFile = File.createTempFile(UUID.randomUUID().toString(), attachment.attachmentType().type());
+      tempFile.deleteOnExit();
+      FileUtils.copyInputStreamToFile(attachment.inputStream(), tempFile);
+    } else {
+      tempFile = null;
+    }
+    return executeAndRetry("send", () -> messagesApi.v4StreamSidMessageCreatePost(
+        streamId,
+        authSession.getSessionToken(),
+        authSession.getKeyManagerToken(),
+        message.content(),
+        message.data(),
+        null,
+        tempFile,
+        null
+    ));
+  }
+
+  /**
+   * Sends a message to the stream ID passed in parameter.
+   *
+   * @param stream    the stream to send the message to
+   * @param message   the message to send to the stream
+   * @return a {@link V4Message} object containing the details of the sent message
+   * @throws IOException if the attachment file cannot be created.
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
+   */
+  public V4Message send(@Nonnull V4Stream stream, @Nonnull Message message) throws IOException {
```

Comment:
with this new method we can avoid adding the other ones sendWithTemplate and other with attachement, can't we?

Reply:
yes indeed

Reply:
we can even remove all the existing sendWithTemplate if you think it's redudant